mobile/native/src/ln_dlc/channel_status.rs

```diff
@@ -0,0 +1,197 @@
+use crate::db::get_all_non_pending_channels;
+use crate::event;
+use crate::ln_dlc::node::Node;
+use anyhow::Result;
+use ln_dlc_node::channel::Channel;
+use ln_dlc_node::channel::ChannelState;
+use ln_dlc_node::node::rust_dlc_manager::subchannel::SubChannel;
+use std::borrow::Borrow;
+use std::time::Duration;
+
+const UPDATE_CHANNEL_STATUS_INTERVAL: Duration = Duration::from_secs(5);
+
+/// The status of the app channels.
+#[derive(Debug, Clone, Copy)]
+pub enum ChannelStatus {
+    /// No channel is open.
+    ///
+    /// This means that it is possible to open a new Lightning channel. This does _not_ indicate if
+    /// there was a previous channel nor does it imply that a previous channel was completely
+    /// closed i.e. there might be pending transactions.
+    NotOpen,
+    /// There is a vanilla Lightning channel, without a subchannel.1
+    ///
+    /// This means that it is possible to use the Lightning channel for payments and that the
+    /// channel can be upgraded to an LN-DLC channel.
+    LnOpen,
+    /// There is an LN-DLC channel.
+    ///
+    /// This corresponds to an app which currently has an open position as a result of trading.
+    LnDlcOpen,
+    // We cannot easily model the vanilla Lightning channel being force-closed because LDK erases
+    // a closed channel from the `ChannelManager` as soon as the closing process begins. This is a
+    // problem because it's not safe to delete the app if the channel is not fully closed!
+    //
+    // TODO: Add support for `LnForceClosing` status and ensure that channel status remains in
+    // `LnDlcForceClosing` whilst the LN channel is still closing.
+    // LnForceClosing,
+    /// The LN-DLC channel is in the process of being force-closed.
+    LnDlcForceClosing,
+    /// The LN-DLC channel is in an inconsistent state.
+    Inconsistent,
+}
+
+pub async fn track_channel_status(node: impl Borrow<Node>) {
```

Comment:
🤔 why not simply: 
```suggestion
pub async fn track_channel_status(node: &Node) {
```
is there any additional benefit for using your construct?

Reply:
AFAIK you can't pass an `Arc` in then.

Reply:
fair enough!  I guess one can also pass in an  `Arc` by doing `track_channel_status(&*node)`, but I see the appeal now.